### PR TITLE
Fix some panics in edge cases.

### PR DESCRIPTION
**Pull request by ehuss**
Fixes a few cases where rustfix will panic.
* An empty file (where there are no lines).
* Span pointing at the end of the file.

### CODE REVIEW

File: tests/parse_and_replace.rs [hunk 1]
```diff
@@ -160,7 +163,7 @@ fn test_rustfix_with_file<P: AsRef<Path>>(file: P, mode: &str) -> Result<(), Err
         ))?;
         let expected_suggestions =
             rustfix::get_suggestions_from_json(&expected_json, &HashSet::new(), filter_suggestions)
-                .context("could not load expected suggesitons")?;
+                .context("could not load expected suggestions")?;
```
Comment (killercup) [inline tests/parse_and_replace.rs:166]: Nice catch :)

File: src/lib.rs [hunk 2]
```diff
@@ -108,8 +108,13 @@ fn parse_snippet(span: &DiagnosticSpan) -> Option<Snippet> {
     let text_slice = span.text[0].text.chars().collect::<Vec<char>>();
 
     // We subtract `1` because these highlights are 1-based
-    let start = span.text[0].highlight_start - 1;
-    let end = span.text[0].highlight_end - 1;
+    // Check the `min` so that it doesn't attempt to index out-of-bounds when
+    // the span points to the "end" of the line. For example, a line of
+    // "foo\n" with a highlight_start of 5 is intended to highlight *after*
+    // the line. This needs to compensate since the newline has been removed
+    // from the text slice.
```
Comment (killercup) [inline src/lib.rs:115]: Oh? Is this a change that the newlines is not part of the text slice?

**Comment by killercup** [comment]
Released as 0.5.1!

File: src/lib.rs (hunk 2 above)
Comment (ehuss) [reply src/lib.rs:115]: AFAIK, rustc has never included line endings in the "text" lines.

It is somewhat unusual for spans to point past the end of the line. `no_main` is one of the few ways to do it.  The `unclosed delimiter` error is the only other way I know of.
